bridges/MastodonBridge.php

```diff
@@ -17,74 +23,142 @@ class MastodonBridge extends FeedExpander {
+			$webfingerHeader = array(
+				'Content-Type: application/jrd+json'
+			);
+			$webfinger = json_decode(getContents($webfingerUrl, $webfingerHeader), true);
+			foreach ($webfinger['links'] as $link) {
+				if ($link['type'] === 'application/activity+json') {
+					return $link['href'];
+				}
+			}
+		}
 
 		return parent::getURI();
 	}
 
-	public function collectData(){
-		return $this->collectExpandableDatas($this->getURI());
+	public function collectData() {
+		$url = $this->getURI() . '/outbox?page=true';
+		$content = json_decode(getContents($url, self::AP_HEADER), true);
+		if ($content['id'] === $url) {
+			foreach ($content['orderedItems'] as $status) {
+				$users = array();
+				$this->items[] = $this->parseItem($status, $users);
+			}
+		} else {
+			throw new \Exception('Unexpected response from server.');
+		}
+	}
+
+	protected function parseItem($content, &$users) {
+		$item = array();
+		switch ($content['type']) {
+			case 'Announce': // boost
+				if ($this->getInput('noboost')) {
+					return null;
+				}
+				// We fetch the boosted content.
+				try {
+					$rtContent = json_decode(getContents($content['object'], self::AP_HEADER), true);
+					if ($rtContent['attributedTo'] && isset($users[$rtContent['attributedTo']])) {
+						$item['author'] = $users[$rtContent['attributedTo']];
+						$item['title'] = 'Shared a status by ' . $item['author'] . ': ';
+						$item = $this->parseObject($rtContent, $item);
+					} else {
+						// We fetch the author, since we cannot always assume the format of the URL.
+						$user = json_decode(getContents($rtContent['attributedTo'], self::AP_HEADER), true);
+						preg_match('/http(|s):\/\/([a-z0-9-\.]{0,})\//', $rtContent['attributedTo'], $matches);
+						// We assume that the server name as indicated by the path is the actual server name,
+						// since using webfinger to delegate domains is not officially supported, and it only
+						// seems to work in one way.
+						$rtUser = '@' . $user['preferredUsername'] . '@' . $matches[2];
+						$users[$rtContent['attributedTo']] = $rtUser;
+						$item['author'] = $rtUser;
+						$item['title'] = 'Shared a status by ' . $rtUser . ': ';
+						$item = $this->parseObject($rtContent, $item);
+					}
+				} catch (UnexpectedResponseException $th) {
+					$item['title'] = 'Shared an unreachable status: ' . $content['object'];
+					$item['content'] = $content['object'];
+					$item['uri'] = $content['object'];
+				}
+				break;
+			case 'Create':
+				if ($this->getInput('norep') && isset($content['object']['inReplyTo'])) {
+					return null;
+				}
+				$item['author'] = $this->getInput('canusername');
+				$item['title'] = '';
+				$item = $this->parseObject($content['object'], $item);
+		}
+		$item['timestamp'] = $content['published'];
+		$item['uid'] = $content['id'];
+		return $item;
+	}
+
+	protected function parseObject($object, $item) {
+		$item['content'] = $object['content'];
+		$strippedContent = strip_tags($object['content']);
```

Comment:
This is good, but also causes problems when there is a `<br>` in the output. Try `@sebsauvage@framapiaf.org`:
![image](https://user-images.githubusercontent.com/8166212/174464838-ce76e61d-e9df-4081-8707-ffef8febbdf9.png)

You could deal with this case separately, excluding `<br>` from strip_tags.